README.md

```diff
@@ -20,47 +20,78 @@
-1) strlen is long enough for the character array timeseries id.
-2) node would be as long as all the nodes of all the timeseries + enough special characters to separate 'holes'. 
-3) timeseries would be as long as the number of timeseries in the file.
-4) index_start_stop would be 2 or 3 depending if just the first node of a given time series or both the first and last node of a given timeseries is to be stored.
-5) time is the length of the maximum length time series in the file. 
+## Data Elements and Structure
+
+For discussion, the structure and attributes below should be what's needed to store a polygon as the `timeseries_id` of a `timeSeries` featuretype. This is using a rectangular array for the time series data and a contiguous ragged array (indexed ragged array would be silly) for the polygon nodes. In CF 2 we would expect the ragged array notation to change to a variable length data field that uses more natural 2d indexing.
+
+### Format overview with timeSeries featureType as context.
+ 
+A piece of software reading this would be expected to:  
+1) See CF-1.8 conventions  
+2) See the timeSeries featureType  
+3) Find the timeseries\_id cf\_role  
+4) Find the coordinates  
+5) See that the coordinates have a standard\_name "polygon node" to determine that these are polygons according to this new specification.  
+6) Find the variable with sample_dimension "node" to determine the count of 'samples' (nodes) along the node dimension. (See CF 1.7 Section 9.3.3, could also use format in 9.3.2 - Incomplete multidimensional array representation.)  
+7) Iterate over polygons, picking out geometries for each `timeSeries` using the count to pick out the required nodes.  
+
+Dimensionality:  
+1) strlen is long enough for the character array timeseries id.  
+2) node would be as long as all the nodes of all the timeseries + enough special characters to separate 'holes'.  
+3) polygons would be as long as the number of timeseries features in the file.  
+4) time is the length of the maximum length time series in the file.  
 
 ```
 netcdf example {
 dimensions:
-        strlen ;
-        node ;
-		timeseries ;
-		index_start_stop ;
-        time ;
+    strlen ;
+    node ;
+    polygons ;
+    time ;
 variables:
-        double polyLat(node) ;
-                polyLat:standard_name = "polygon node" ;
-        double polyLon(node) ;
-                polyLon:standard_name = "polygon node" ;
-		int polyIndex(timeseries, index_start_stop) ;
-				polyIndex:cf_role = "node_index" ;
-        char timeseriesID(timeseries, strlen) ;
-                timeseriesID:cf_role = "timeseries_id" ;
-        int data(timeseries, time) ;
-                data:coordinates = "time polyLat polyLon" ;
+    int crs() ;
+    double polyLat(node) ;
+				polyLat:standard_name = "polygon y node" ;
+				polyLat:grid_mapping = "crs"
+    double polyLon(node) ;
+				polyLon:standard_name = "polygon x node" ;
+				polyLon:grid_mapping = "crs"
+    int polyIndex(polygons) ;
```

Comment:
You asked for suggestions. Here's one:


A general system for Contiguous Ragged Arrays that mimics the DSG CRA but
uses a different attribute name, applied here is:

int count(timeseries);
  contiguous_ragged_dimension = "obs" ;
double polyLat(obs);
double polyLon(obs);

* The variable names don't matter.
* The attribute "contiguous_ragged_dimension" and its value
  (the name of a dimension) are the key.
  Any var that uses that named dimension is to be interpreted using the
  counts (in this case, the number of values for each timeseries feature)
  in variable that has the "contiguous_ragged_dimension" for that dimension.
* With a few rules, this works for multidimensional 'count' variables.
* Then we have a system in CF 1.0/netcdf-3 to parallel vlen
dimensions/arrays in CF 2.0/netcdf-4.

---
Currently, CF has no String data type and it is ambiguous whether an
n-dimensional char array should be interpreted as an n-dimensional char
array or an (n-1)-dimensional String.  We can make a variant of the above
system with a different attribute name to add support for a vlen String
data type:

int count(timeseries);
  string_dimension = "charDim" ;
char comment(charDim);

* Again, the attribute name "string_dimension" and its value are the key.
  For "string_dimension" the implication is: these char arrays are
  to be converted to Strings of varying length.
* Then we have system in CF 1.0/netcdf-3 system to parallel vlen Strings in
CF 2.0/netcdf-4



On Tue, Jun 7, 2016 at 12:44 PM, David Blodgett <notifications@github.com>
wrote:

> In README.md
> <https://github.com/bekozi/netCDF-CF-simple-geometry/pull/13#discussion_r66138986>
> :
>
> > -        double polyLon(node) ;
> > -                polyLon:standard_name = "polygon node" ;
> > -		int polyIndex(timeseries, index_start_stop) ;
> > -				polyIndex:cf_role = "node_index" ;
> > -        char timeseriesID(timeseries, strlen) ;
> > -                timeseriesID:cf_role = "timeseries_id" ;
> > -        int data(timeseries, time) ;
> > -                data:coordinates = "time polyLat polyLon" ;
> > +    int crs() ;
> > +    double polyLat(node) ;
> > +				polyLat:standard_name = "polygon y node" ;
> > +				polyLat:grid_mapping = "crs"
> > +    double polyLon(node) ;
> > +				polyLon:standard_name = "polygon x node" ;
> > +				polyLon:grid_mapping = "crs"
> > +    int polyIndex(polygons) ;
>
> Hmmm... it's still an index of sorts. The semantics here are going to be
> cumbersome for sure. Open to suggestions.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bekozi/netCDF-CF-simple-geometry/pull/13/files/9e6a2ffdadd54ec4b128f6b31232dc208154dcad#r66138986>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABarOMO-cC7YDRJDqLo0oG2RxUZ6q8N1ks5qJcopgaJpZM4IuIL1>
> .
>



-- 
Sincerely,

Bob Simons
IT Specialist
Environmental Research Division
NOAA Southwest Fisheries Science Center
99 Pacific St., Suite 255A      (New!)
Monterey, CA 93940               (New!)
Phone: (831)333-9878            (New!)
Fax:   (831)648-8440
Email: bob.simons@noaa.gov

The contents of this message are mine personally and
do not necessarily reflect any position of the
Government or the National Oceanic and Atmospheric Administration.
<>< <>< <>< <>< <>< <>< <>< <>< <><